hatch_build.py

```diff
@@ -53,3 +57,72 @@ def finalize(
         Clean up any generated temporary files.
         """
         self.known_implementations().unlink()
+
+    def _known_implementations_from_packages_api(
+        self,
+        known_local: list[str],
+    ) -> list[str]:
+        """
+        Collects available implementation using GitHub packages API.
+        """
+        gh_token = os.getenv("GITHUB_TOKEN")
+
+        if not gh_token:
+            self.app.display_warning(
```

Comment:
(I'm also OK immediately requiring the token to be set, as really the only thing that runs this script is GitHub actions here in the repo, and me when building the package locally I think) But this is of course good as-is.

Reply:
Yeah that seems like a nice idea too!